File: platform-core/src/main/java/com/webank/wecube/platform/core/domain/SystemVariable.java
//package com.webank.wecube.platform.core.domain;
//
//import javax.persistence.*;
//
//import com.webank.wecube.platform.core.support.DomainIdBuilder;
//import org.apache.commons.lang.builder.ReflectionToStringBuilder;
//
//@Entity
//@Table(name = "system_variables")
//public class SystemVariable {
//    public static final String ACTIVE = "active";
//    public static final String INACTIVE = "inactive";
//
//    public static final String SCOPE_GLOBAL = "global";
//    public static final String SOURCE_SYSTEM = "system";
//
//    @Id
//    private String id;
//
//    @Column
//    private String scope;
//
//    @Column
//    private String packageName;
//
//	@Column
//	private String name;
//	@Column
//	private String value;
//	@Column
//	private String defaultValue;
//
//	@Column
//    private String source;
//
//	@Column
//	private String status;
//
//	public String getId() {
//		return id;
//	}
//	
//    public void setId(String id) {
//        this.id = id;
//    }
//
//    @PrePersist
//    public void initId() {
//        this.id = DomainIdBuilder.buildDomainId(this);
//    }
//
//    public String getPackageName() {
//        return packageName;
//    }
//
//    public void setPackageName(String packageName) {
//        this.packageName = packageName;
//    }
//
//    public String getScope() {
//        return scope;
//    }
//
//    public void setScope(String scope) {
//        this.scope = scope;
//    }
//
//
//    public String getName() {
//        return name;
//    }
//
//    public void setName(String name) {
//        this.name = name;
//    }
//
//    public String getValue() {
//        return value;
//    }
//
//    public void setValue(String value) {
//        this.value = value;
//    }
//
//    public String getDefaultValue() {
//        return defaultValue;
//    }
//
//    public void setDefaultValue(String defaultValue) {
//        this.defaultValue = defaultValue;
//    }
//
//    public String getStatus() {
//        return status;
//    }
//
//    public void setStatus(String status) {
//        this.status = status;
//    }
//
//    public String getSource() {
//        return source;
//    }
//
//    public void setSource(String source) {
//        this.source = source;
//    }
//
//    public SystemVariable(String id, String name, String value, String defaultValue, String scope,
//                          String source, String packageName, String status) {
//        this.id = id;
//        this.name = name;
//        this.value = value;
//        this.defaultValue = defaultValue;
//        this.scope = scope;
//        this.source = source;
//        this.packageName = packageName;
//        this.status = status;
//    }
//
//    public SystemVariable() {
//        super();
//    }
//
//    public SystemVariable activate() {
//	    this.status = SystemVariable.ACTIVE;
//	    return this;
//    }
//
//    public SystemVariable deactivate() {
//	    this.status = SystemVariable.INACTIVE;
//	    return this;
//    }
//
//    @Override
//    public String toString() {
//        return ReflectionToStringBuilder.toString(this);
//    }
//
//}

File: platform-core/models/system-variable.go
package models

const (
	SystemVariableInactive  = "inactive"
	SystemVariableActive    = "active"
	ScopeGlobal             = "global"
	SourceSystem            = "system"
	SystemVariableUmContext = "UM_AUTH_CONTEXT"
)

type SystemVariables struct {
	Id           string `json:"id" xorm:"id"`                      // 唯一标识
	PackageName  string `json:"packageName" xorm:"package_name"`   // 包名
	Name         string `json:"name" xorm:"name"`                  // 变量名
	Value        string `json:"value" xorm:"value"`                // 变量值
	DefaultValue string `json:"defaultValue" xorm:"default_value"` // 默认值
	Scope        string `json:"scope" xorm:"scope"`                // 作用范围
	Source       string `json:"source" xorm:"source"`              // 来源
	Status       string `json:"status" xorm:"status"`              // 状态 -> active | inactive
}

type SystemVariablesListPageData struct {
	PageInfo *PageInfo          `json:"pageInfo"` // 分页信息
	Contents []*SystemVariables `json:"contents"` // 列表内容
}

type SystemVariablesQueryCondition struct {
	Name   string `json:"name"`   // 变量名
	Scope  string `json:"scope"`  // 作用范围
	Status string `json:"status"` // 状态 -> active | inactive
}
